### PR TITLE
Feature/test levels

### PR DESCRIPTION
**Pull request by xross**
Added proper parameterisation to tests.

Run with --smoke for minimal param set (Pytest --smoke)

All smoke tests currently pass or marked fail - all xfailed tests should have an issue

### CODE REVIEW

File: tests/test_iso_tx_basic/src/main.xc [hunk 1]
```diff
@@ -1,61 +1,54 @@
...
+                                                XUD_EPTYPE_ISO,
+                                                XUD_EPTYPE_ISO,
                                                 XUD_EPTYPE_ISO};
 
 int main()
 {
-    chan c_ep_out[XUD_EP_COUNT_OUT], c_ep_in[XUD_EP_COUNT_IN];
-    chan c_dummy[DUMMY_THREAD_COUNT];
+    chan c_ep_out[EP_COUNT_OUT], c_ep_in[EP_COUNT_IN];
+    //chan c_dummy[DUMMY_THREAD_COUNT];
```
Comment (DaveAtkinsonXMOS) [inline tests/test_iso_tx_basic/src/main.xc:31]: Any reason to keep the commented-out stuff?

File: tests/test_iso_tx_basic/src/main.xc [hunk 2]
```diff
@@ -1,61 +1,54 @@
...
 		{
-			TestEp_Tx(c_ep_in[3], 3, 10, 14, RUNMODE_DIE);
+			TestEp_Tx(c_ep_in[TEST_EP_NUM], TEST_EP_NUM, 10, 14, RUNMODE_DIE);
 			XUD_ep ep_out_0 = XUD_InitEp(c_ep_out[0]);
 			XUD_Kill(ep_out_0);
 
-            for(size_t i = 0; i<DUMMY_THREAD_COUNT; i++)
-                outuint(c_dummy[i], 0);
+            //for(size_t i = 0; i<DUMMY_THREAD_COUNT; i++)
+            //    outuint(c_dummy[i], 0);
```
Comment (DaveAtkinsonXMOS) [inline tests/test_iso_tx_basic/src/main.xc:46]: Any reason to keep the commented-out stuff?

File: tests/test_iso_tx_basic/src/main.xc [hunk 3]
```diff
@@ -1,61 +1,54 @@
...
 			
             exit(0);
 		}
         
-        dummyThreads(c_dummy);
+        //dummyThreads(c_dummy);
```
Comment (DaveAtkinsonXMOS) [inline tests/test_iso_tx_basic/src/main.xc:51]: Any reason to keep the commented-out stuff?

**Review by DaveAtkinsonXMOS — APPROVED** [review]
Usual deal - I found some commented-out code and a piece of un-Pythonic Python (that isn't even part of the PR). Change or merge it, they're not show stoppers.

File: tests/Pyxsim/__init__.py [hunk 4]
```diff
@@ -51,6 +51,8 @@ def _build(xe_path, build_config=None, env={}, do_clean=False):
     if build_config != None:
```
[the remark this answers is not in the document]
Comment (xross) [reply tests/Pyxsim/__init__.py:51]: Fixed.

File: tests/test_iso_tx_basic/src/main.xc (hunk 1 above)
Comment (xross) [reply tests/test_iso_tx_basic/src/main.xc:31]: This needs re-instating (as per issue #114). However, the current scheme fails due to running out of channel ends. Need to come up with a different scheme.

**Pull request merged**